mqtt/mqtt-bridge/src/client.rs

```diff
@@ -281,13 +319,16 @@ impl<H: EventHandler> MqttClient<H> {
             keep_alive,
         );
 
+        let in_flight = Arc::new(Semaphore::new(max_in_flight));
```

Comment:
I think instead of storing semaphore, it can store InFlighPublishHandle. Return self.publish_handle copy in the `fn publish_handle`()`

Reply:
Creating the client's publish_handle can error. It might be better to see this potential error every time we get the client wrapper's publish handle. 

I don't think it will matter though. Even when we recreate the pump it will recreate the client and only get the publish handle once at the beginning.